### PR TITLE
atlas-memory: bridge runtime substrate in core

### PR DESCRIPTION
**Pull request by OrionArchitekton**
## Summary
- implement PR-3C for `M-012` using the admitted `M-013` runtime baseline
- switch Atlas search/write/namespaces in `cosmocrat-core` to prefer the `orion_runtime.atlas_memory` seam
- keep archive/offload/restore, startup wiring, and other out-of-scope Atlas surfaces in core

## Scope
- row-bound to `M-012`
- paired with OrionArchitekton/orion-runtime#45
- no work on `M-015`, `M-016`, `M-017`, archive metadata, infra packaging, or governance repos

## What changed
- `app/services/atlas-memory/main.py` now prefers the runtime executor seam for internal search/write/namespaces
- `app/services/edge_orchestrator/atlas_memory_proxy.py` now prefers the runtime proxy seam for search/write/namespaces
- archive transition methods remain in core and were not moved
- added hermetic bridge coverage plus runtime/core parity verification

## Verification
- `python3 -m py_compile app/services/edge_orchestrator/atlas_memory_proxy.py app/services/atlas-memory/main.py app/services/memory/tests_hermetic/test_runtime_atlas_substrate_bridge.py`
- `TMPDIR=/home/orion/tmp-pytest python3 -m pytest --capture=no tests/ops/test_runtime_atlas_memory_bridge.py app/services/memory/tests_hermetic/test_runtime_atlas_substrate_bridge.py tests/governance/test_memory_search_tenant_tripwire.py tests/governance/test_quick_wins.py -k 'AtlasMemoryProxy or atlas_internal_search or runtime_atlas_memory_bridge or atlas_main'`
- `TMPDIR=/home/orion/tmp-pytest PYTHONPATH=/tmp/orion-runtime-pr3c/src:. python3 -m pytest --capture=no tests/governance/test_quick_wins.py -k 'AtlasMemoryProxy'`
- `python3 ops/scripts/check_runtime_atlas_memory_bridge.py --repo-root . --runtime-root /tmp/orion-runtime-pr3c`

Track-ID: cleanup

## Summary by Sourcery

Prefer the Orion runtime Atlas memory substrate for internal search, write, and namespace operations while keeping legacy core paths as a fallback and preserving archive-related flows in core.

New Features:
- Introduce an optional runtime-backed Atlas memory proxy in the edge orchestrator that takes precedence when the runtime module is available.
- Introduce an optional runtime-backed Atlas memory executor in the atlas-memory service to handle internal search, write, and namespace listing when present.

Enhancements:
- Adapt existing HTTP transport and envelope serialization/deserialization to interoperate with the runtime Atlas memory APIs without changing external contracts.
- Update the tracked orion-runtime Atlas memory substrate commit to align with the new bridge integration.

Tests:
- Add hermetic tests to verify that the edge Atlas memory proxy and atlas-memory service prefer the runtime substrate when available and maintain parity with core behavior.

### CODE REVIEW

**Comment by OrionArchitekton** [comment]
Architecture review — PR #558 (PR-3C core Atlas bridge consumer)

Verdict: This is the correct companion PR to PR #45 and is merge-worthy as the core-side compatibility bridge for PR-3C.

What this PR does correctly:
- It keeps `cosmocrat-core` in bridge posture only.
- It switches Atlas search/write/namespaces to prefer the `orion_runtime.atlas_memory` seam.
- It leaves archive/offload/restore, startup wiring, and other out-of-scope Atlas surfaces in core.
- It stays limited to the PR-3C row set for `M-012`.

Boundary read:
- runtime remains the Atlas substrate home
- core remains bridge-only for these paths
- archive semantics remain outside runtime and outside this extraction
- `M-013` remains the admitted baseline seam, not a new broad landing zone

Required sequencing rule:
- do not treat this PR as independently completing PR-3C
- it is the bridge counterpart to PR #45 and depends on the runtime seam introduced there
- if merge ordering matters, prefer runtime first, then core bridge

What is correct and should remain true after merge:
- no archive transitions moved into runtime-facing bridge code
- no receipt durability or mutation authority drift into runtime or back into core authority paths
- no new `httpx`, FastAPI ownership, or store ownership moved into runtime via this bridge
- no widening into `M-015`, `M-016`, `M-017`, infra, governance, or retrieval platform work

Follow-on impact:
- after PR #45 and PR #558 land, PR-3C is complete for `M-012`
- this does not authorize work on `M-015`, `M-016`, or blocked rows

Review guardrails:
- reject any follow-on PR that expands these bridge modules into permanent archive, startup, or orchestration surfaces
- reject any attempt to treat `M-016` as now movable ahead of `M-015`
- reject any scope expansion beyond the cited PR-3C row

Bottom line:
- PR #558 is the correct core bridge companion for PR-3C
- merge-worthy as bridge-only
- should be treated as temporary compatibility infrastructure tied to PR #45 and the admitted `M-013` seam

File: app/services/edge_orchestrator/atlas_memory_proxy.py [hunk 1]
```diff
@@ -41,6 +42,67 @@
...
+class _RuntimeEnvelopeSerializer:
+    """Serialize legacy core envelopes for the runtime Atlas proxy bridge."""
+
+    def serialize(self, envelope: object) -> Optional[Dict[str, Any]]:
+        try:
+            return envelope.model_dump()
+        except Exception:
+            if hasattr(envelope, "dict"):
+                return envelope.dict()
+            return None
```
[the remark this answers is not in the document]
Comment (OrionArchitekton) [reply app/services/edge_orchestrator/atlas_memory_proxy.py]: Fixed in 7b82bd7. The runtime envelope serializer no longer swallows broad exceptions; it now does the version-compatibility check explicitly (`model_dump` first, then `dict`) and lets unexpected serialization failures surface.

File: app/services/edge_orchestrator/atlas_memory_proxy.py [hunk 2]
```diff
@@ -306,7 +429,67 @@ async def write(
...
+                        "trace_id": trace_id,
+                    },
+                )
+                if status_code in (400, 403, 409):
+                    parsed_detail: Any = raw_detail
+                    try:
+                        payload = e.response.json()
+                        parsed_detail = payload.get("detail", payload) if isinstance(payload, dict) else payload
+                    except Exception:
+                        parsed_detail = raw_detail
```
[the remark this answers is not in the document]
Comment (OrionArchitekton) [reply app/services/edge_orchestrator/atlas_memory_proxy.py]: Fixed in 7b82bd7. The JSON detail fallback now catches `ValueError`, which is the expected parse failure path for `response.json()`, instead of swallowing unrelated exceptions.
